cli/src/cast.rs

```diff
@@ -710,6 +728,51 @@ where
     })
 }
 
+fn det_base_in(value: &str, base_in: Option<String>) -> eyre::Result<u32> {
+    match base_in {
+        Some(base_in) => match base_in.as_str() {
+            "10" | "dec" => Ok(10),
+            "16" | "hex" => Ok(16),
+            _ => Err(eyre::eyre!("could not parse hex base in")),
+        },
+        None => {
+            if value.starts_with("0x") {
+                U256::from_str_radix(value, 16).expect("could not parse hex base in");
+
+                Ok(16)
+            } else {
```

Comment:
Using the other pattern change, we can change this to

```suggestion
        None => {
```